client/my-sites/plugins/index.web.js

```diff
@@ -31,11 +43,12 @@ export default function () {
 		clientRender
 	);
 
-	page(
-		'/plugins/browse/:category/:site?',
+	router(
+		`/${ langParam }/plugins/browse/:category/:site(${ siteId })?`,
+		redirectWithoutLocaleParamIfLoggedIn,
 		scrollTopIfNoHash,
 		siteSelection,
-		navigation,
+		navigationIfLoggedIn,
```

Comment:
I had to go back on the suggestion to have [separate routes for logged in / logged out users](https://github.com/Automattic/wp-calypso/pull/66087/files#r934127573) because there's a route that can be accessed in both situations: `/plugins`. 
I tried separating based on if there's a site selected (as in /themes) but things got a bit hairy and couldn't find a proper workaround for the `/plugins` route.

Using this approach also addresses an issue where the Navigation CSS was not loaded properly when directly accessing the Plugins Detail page.

Reply:
Didn't take a close enough look apparently, a lot of these are localized but shouldn't be. Logged out users don't need access to most of these logged in only routes. Doesn't seem to be any upside to having these localized, is there any downsides to leaving them as is?

Made an issue to fix, let me know if there are any issues with that. https://github.com/Automattic/wp-calypso/issues/67711

Reply:
>  Doesn't seem to be any upside to having these localized, is there any downsides to leaving them as is?

I agree but we'll still need to keep the `/${ langParam }` and `redirectWithoutLocaleParamIfLoggedIn` for unlocalized routes. 
Otherwise, these routes will end up being caught by the Plugin details route: `/${ langParam }/plugins/:plugin/:site_id?`

To avoid localizing the logged-in routes, we'll need to specify them `index.node.js` file.

Reply:
Can you explain how this works a bit more? We just added this route: https://github.com/Automattic/wp-calypso/pull/67710/files?w=1#diff-7edd3c450e9fa353a0acd99fdf2b07d0c902ef5c88f419a3f9ddd3df242822f8R92

Reply:
For example, while logged-in and accessing a link such as `/ja/plugins/plans/{SITE}`:
* first it gets caught by the plugins details route `/${ langParam }/plugins/:plugin/:site_id?`
* `redirectWithoutLocaleParamIfLoggedIn` redirects to `/plugins/plans/{SITE}`
* `/plugins/plans/:site` now correctly catches the request

Other scenarios such as accessing `/ja/plugins/plans/{SITE}` as logged-out is also caught by the plugin details route but that also redirects to login because of the `{SITE}` in the URL. 
If `{SITE}` is not present, then it'll show an empty/placeholder details page as it can't find the `plans` plugin.

Reply:
Ah didn't see langParam was optional or how that redirect was working, gotcha, thanks.